sdcm/cluster_aws.py

```diff
@@ -91,11 +90,21 @@ def __init__(self, ec2_ami_id, ec2_subnet_id, ec2_security_group_ids,
             ec2_block_device_mappings = []
         self._ec2_block_device_mappings = ec2_block_device_mappings
         self._ec2_user_data = ec2_user_data
-        self._ec2_ami_id = ec2_ami_id
         self.region_names = region_names
-        self.instance_provision = params.get('instance_provision', default=INSTANCE_PROVISION_ON_DEMAND)
         self.params = params
         self.node_type = node_type
+
+        # try to convert AMI-name to AMI-ID
+        new_ec2_ami_id = []
+        for idx, ec2_ami in enumerate(ec2_ami_id):
+            ec2 = ec2_client.EC2Client(region_name=self.region_names[idx],
+                spot_max_price_percentage=self.params.get('spot_max_price', default=0.60))
```

Comment:
`EC2Client` should have default value for spot_max_price_percentage` in the class. We should store defaults in one place.

Reply:
remove this code.